linode/manager
# refactor: [M3-6260] - MUI v5 Migration - SRC > Features > Backups

## Description 📝
- MUI style migration for Backups feature

## Major Changes 🔄
### BackupsTable
- [x] Remove React.FC and rename interface 'Props' to '{ComponentName}Props'  
- [x] You can remove 'CombinedProps' if it's directly referencing Props with nothing additional  
- [x] Export component inline, and remove 'export default {ComponentName}' at bottom of file. You will have to fix the exports/imports throughout the project where it's used  
- [x] Any style attributes can be converted to 'sx prop' or 'styled API'

### BackupsCTA
- [x] Remove React.FC and rename interface 'Props' to '{ComponentName}Props'  
- [x] Remove makeStyles for new styled API. We typically use 'Styled ComponentName) as our prefix  
- [x] Export component inline, and remove 'export default {ComponentName}' at bottom of file. You will have to fix the exports/imports throughout the project where it's used  
- [x] Any style attributes can be converted to 'sx prop' or 'styled API'
- [x] The `button` element can probably use `components/Button/StyledLinkButton.ts`
- [x] The close `button` can probably use `import { IconButton } from 'src/components/IconButton';`

### BackupsLinode
- [x] Remove React.FC and rename interface 'Props' to '{ComponentName}Props'  
- [x] Export component inline, and remove 'export default {ComponentName}' at bottom of file. You will have to fix the exports/imports throughout the project where it's used  
- [x] Any style attributes can be converted to 'sx prop' or 'styled API'

### AutoEnroll
- [x] Remove React.FC and rename interface 'Props' to '{ComponentName}Props'   
- [x] Export component inline, and remove 'export default {ComponentName}' at bottom of file. You will have to fix the exports/imports throughout the project where it's used  
- [x] Any style attributes can be converted to 'sx prop' or 'styled API'

### BackupDrawer
- [x] Remove React.FC and rename interface 'Props' to '{ComponentName}Props'  
- [x] Export component inline, and remove 'export default {ComponentName}' at bottom of file. You will have to fix the exports/imports throughout the project where it's used  
- [x] Any style attributes can be converted to 'sx prop' or 'styled API'
- [x] Convert this to a functional component

## Preview 📷
Before:
![Screenshot 2023-07-21 at 1 33 26 PM](https://github.com/linode/manager/assets/139489745/93fe001a-5e6d-41c0-9dad-8472a25ff98b)

![Screenshot 2023-07-21 at 1 32 09 PM](https://github.com/linode/manager/assets/139489745/45f2d666-5fa3-446c-b132-2a2011491696)

After:
![Screenshot 2023-07-21 at 1 32 51 PM](https://github.com/linode/manager/assets/139489745/193d229e-9f76-4422-9971-03e9a9cff5a1)

![Screenshot 2023-07-21 at 1 31 38 PM](https://github.com/linode/manager/assets/139489745/6e309a7f-8819-4059-a4b5-49a7f991adda)


## How to test 
Navigate to the Linodes page. There should be a popup on the top of the screen that says "Enable Linode Backups to protect your data and recover quickly in an emergency."
![Screenshot 2023-07-21 at 1 11 01 PM](https://github.com/linode/manager/assets/139489745/9b8c9cd4-2848-416e-b13e-321bb5324657)

If you dont have it:
This means you previously clicked 'x' so it doesn't pop-up anymore. An easy work around is to go into BackupsCTA.tsx line 49, and change "isDismissed" to "!isDismissed". The pop-up should now be there. Hit "x" one more time in the pop-up, and go back and get rid of the '!' before isDismissed. This is the only work around I have found so far. I haven't been able to find a way to bring the pop-up back within Cloud Manager.

Make sure the side panel pops up and you can sign up for backups for the correct Linodes. Also make sure the styling is consistent with the dev branch.
